File: spring-integration/subscribable-channels/direct-channel/src/main/java/com/ddubson/integration/services/PrintService.java
package com.ddubson.integration.services;

import com.ddubson.ANSIColor;
import com.ddubson.LogAdapter;
import org.springframework.messaging.Message;

import static java.lang.String.format;

public class PrintService {
    private final LogAdapter logAdapter;

    public PrintService(LogAdapter logAdapter) {
        this.logAdapter = logAdapter;
    }

    public void print(Message<String> msg) {
        logAdapter.info(
                format("[2] Consuming input channel -- [%s] %s", Thread.currentThread().getName(),
                        msg.getPayload()), ANSIColor.ANSI_GREEN);
    }
}

File: spring-integration/subscribable-channels/direct-channel/src/main/kotlin/com/ddubson/integration/services/PrintService.kt
package com.ddubson.integration.services

import com.ddubson.ANSIColor
import com.ddubson.LogAdapter
import org.springframework.messaging.Message

class PrintService(private val logAdapter: LogAdapter) {
    fun print(msg: Message<String>) {
        logAdapter.info("[2] Consuming input channel -- [${Thread.currentThread().name}] ${msg.payload}", ANSIColor.ANSI_GREEN)
    }
}
